data/virt_autotest/clean_up_virt_logs.sh

```diff
@@ -0,0 +1,173 @@
+#!/bin/bash
+
+#Determine whether log file is text file
+function is_text_file() {
+	local log_file=$1
+
+	if [[ `file ${log_file} | grep ":.*text"` ]];then
+	   echo "YES"
+	else
+	   echo "NO"
+	fi
+}
+
+#Determine whether log file is empty file
+function is_empty_file() {
+	local log_file=$1
+
+	if [[ `file ${log_file} | grep ":.*empty"` ]];then
+	   echo "YES"
+	else
+	   echo "NO"
+	fi
+}
+
+#Empty text log file or rm empty text log file
+function cleanup_single_file() {
+	local file_path=$1
+
+	is_text=`is_text_file ${file_path}`
+	is_empty=`is_empty_file ${file_path}`
+	if [[ ${is_text} == "YES" ]];then
+	   > ${file_path}
+	   echo -e "Cleaned up text log file ${file_path}."
+	   return 1
+	elif [[ ${is_empty} == "YES" ]];then
+	   rm -f -r ${file_path}
+	   echo -e "Removed empty log file ${file_path}."
+	   return 1
+	else
+	   return 0
+	fi
+}
+
+#Clean log files or folders by using recursive algorithm
+function cleanup_logs() {
+	local logs_path=$1
+	local total_text_file=0
+	local is_text=""
+	local is_empty=""
+	local eachlog=""
+
+	if [[ -f ${logs_path} ]];then
```

Comment:
'ls -R $logs_path' will list all  in the directory recursively, with a for loop on it, this function will be much simpler

Reply:
But it does not list all sub-folders in plain format. It will increase complexity in post-processing. It, at least, needs another two more for loops. Compared to that, recursive algorithm is much easier and compact.

Reply:
You should know that "ls -R" is just a method to list, it does not solve problem. We use recursive function to solve the real problem. It lets you see everything inside, but does not solve the problem, namely how to clean up those folders or files, for you. At the last, you need to resort some kind of algorithm. There is no other ways to cut short.